test/services/tempMailApi.test.js

```diff
@@ -35,6 +37,78 @@ describe('TempMail', () => {
       });
     });
 
+    it('should throw err if domain is null or cannot retrieve it', async () => {
+      axios.get.mockRejectedValueOnce(new Error());
+      try {
+        await tempMailTest.createMail({username: 'enElSiguientePr', password: '1111'});
+      } catch (error) {
+        expect(axios.get).toHaveBeenCalledTimes(1);
+        expect(axios.get).toHaveBeenCalledWith('https://api.mail.tm/domains');
+
+        expect(error).toBeInstanceOf(InvalidMailCreationError);
+        expect(error.message).toEqual('Cant create new mail.');
+      }
+    });
+
+
+    it('should throw err if domain/username or credetials are not valid', async () => {
+      axios.get
+          .mockReturnValueOnce({
+            data: {
+              'hydra:member': [{domain: 'patata.net'}],
+            },
+          });
+
+      axios.post.mockRejectedValueOnce(new Error());
+      try {
+        await tempMailTest.createMail({username: 'enElSiguientePr', password: '1111'});
+      } catch (error) {
+        expect(axios.get).toHaveBeenCalledTimes(1);
+        expect(axios.get).toHaveBeenCalledWith('https://api.mail.tm/domains');
+        expect(axios.post).toHaveBeenCalledTimes(1);
+        expect(axios.post).toHaveBeenCalledWith('https://api.mail.tm/accounts', {
+          address: 'enElSiguientePr@patata.net',
+          password: '1111',
+        });
+        expect(error).toBeInstanceOf(InvalidMailCreationError);
+        expect(error.message).toEqual('Cant create new mail.');
+      }
+    });
+
+    it('should throw err if username is already on use', async () => {
```

Comment:
Why you're one is returnvalue and the other is reject value?

you need both?

Reply:
I did understand the flow of the test should work as expected as the real method does, so i want to emulate the by getting the domain, returning me the domain and failing when the username is repeated, that was my idea, of using both